glass/shells.py

```diff
@@ -195,16 +196,31 @@ class RadialWindow(NamedTuple):
     zeff
         Effective redshift of the window.
 
-    Methods
-    -------
-    _replace
-        Create a new instance with changed attribute values.
-
     """
 
     za: NDArray[np.float64]
     wa: NDArray[np.float64]
-    zeff: float = 0
+    zeff: float | None = dataclasses.field(default=None)
+
+    def __post_init__(self) -> None:
+        """Magic method to calculate the effective redshift if not given."""
+        if self.zeff is None:
+            object.__setattr__(self, "zeff", self._calculate_zeff())
+
+    def _calculate_zeff(self) -> float:
+        """Calculate ``zeff`` if not given.
+
+        Returns
+        -------
+            The effective redshift depending on the size of ``za``.
+
+        """
+        if self.za.size > 0:
+            return np.trapezoid(  # type: ignore[return-value]
```

Comment:
Are we certain that the expression will never return `ndarray`?

Reply:
I guess it depends on what `za` and `wa` are expected to be? My feeling is 1-D arrays, but might be wrong

Reply:
Yes, I think these should be 1D arrays always.